pkg/pillar/zedcloud/authen.go

```diff
@@ -88,6 +89,58 @@ func RemoveAndVerifyAuthContainer(ctx *ZedCloudContext,
 	return nil
 }
 
+func decryptCipherBlock(decryptCtx *cipher.DecryptCipherContext,
+	sm *zauth.AuthContainer, cipherBlock *zcommon.CipherBlock) ([]byte, error) {
+	cipherContext := sm.GetCipherContext()
+	if decryptCtx == nil {
+		err := errors.New("removeAndVerifyAuthContainer: decrypt cipher context is undefined\n")
+		return nil, err
+	}
+	if cipherContext == nil {
+		err := errors.New("removeAndVerifyAuthContainer: cipher context is undefined\n")
+		return nil, err
+	}
+	if len(cipherBlock.CipherData) == 0 ||
+		len(cipherBlock.CipherContextId) == 0 {
+		err := errors.New("removeAndVerifyAuthContainer: cipher block data or context id are incorrect\n")
+		return nil, err
+	}
+	if cipherContext.ContextId != cipherBlock.CipherContextId {
+		err := errors.New("removeAndVerifyAuthContainer: cipher context ids do not match\n")
+		return nil, err
+	}
+	cipherCtx := &types.CipherContext{
+		ContextID:          cipherContext.GetContextId(),
+		HashScheme:         cipherContext.GetHashScheme(),
+		KeyExchangeScheme:  cipherContext.GetKeyExchangeScheme(),
+		EncryptionScheme:   cipherContext.GetEncryptionScheme(),
+		DeviceCertHash:     cipherContext.GetDeviceCertHash(),
+		ControllerCertHash: cipherContext.GetControllerCertHash(),
+	}
+
+	// CRC32-Q, copied from the hash/crc32 manual
+	crc32q := crc32.MakeTable(0xD5828281)
+	// Generate a signature-dependent key
+	key := fmt.Sprintf("%s-%d", cipherBlock.GetCipherContextId(),
+		crc32.Checksum(sm.GetSignatureHash(), crc32q))
```

Comment:
Why isn't the cipherContextId sufficiently unique for this?
I don't think you are publishing this using pubsub so does the CipherBlockID matter at all?

Reply:
I was not sure, do we need this key or not. So decided to be on the safe side. I'm not publishing, so probably not needed.

Reply:
If you decide to keep the code you should add a comment about that. But it seems better to validate that it isn't needed.